lib/hpc/thread1.g

```diff
@@ -168,6 +175,43 @@ BIND_GLOBAL("CopyToRegion", {x,y} -> x);
 BIND_GLOBAL("SetTLDefault", BindThreadLocal);
 BIND_GLOBAL("SetTLConstructor", BindThreadLocalConstructor);
 
+
+BIND_GLOBAL("CHECKED_GET_ATOMIC_LIST", function(l, pos)
+    if IsBound(l[pos]) then
+        return [true, l[pos]];
+    else
+        return [false];
+    fi;
+end);
+
+BIND_GLOBAL("COMPARE_AND_SWAP", function(l, pos, old, new)
+    if IsBound(l[pos]) and IS_IDENTICAL_OBJ(l[pos], old) then
+        l[pos] := new;
+        return true;
+    else
+        return false;
+    fi;
+end);
+
+BIND_GLOBAL("ATOMIC_BIND", function(l, pos, new)
+    if IsBound(l[pos]) then
+        return false;
+    else
+        l[pos] := new;
+        return true;
+    fi;
+end);
+
+BIND_GLOBAL("ATOMIC_UNBIND", function(l, pos, old)
+    if not (IsBound(l[pos]) and IS_IDENTICAL_OBJ(l[pos], old)) then
```

Comment:
How about getting rid of the `not` and swapping the if/else cases? Just as in `COMPARE_AND_SWAP`